### ISSUE TITLE
Misleading index argument in DataFrame constructor

### ISSUE THREAD

**Issue by floscha** [databricks/koalas#468]
When creating a new DataFrame with Koalas, the class' constructor accepts an `index` argument just like pandas. Unlike for pandas, however, `index` can't be a list of scalars but only a native `Metadata` object which is really confusing for pandas users. While I understand the reason `Metadata` exists, I'd suggest renaming the existing `index` argument to `metadata` and instead have an `index` argument which accepts a list of scalars and automatically infers `data_columns` like when calling the constructor with `metadata=None`.

**Comment by ueshin** [comment]
Could you elaborate what usage you are thinking about?
The constructor similar usage to `pd.DataFrame` should work fine with `index` argument as well.

Btw, I'm working on refactoring using `_InternalFrame` and `Metadata` will be removed.

**Comment by thunterdb** [comment]
To simplify, `pd.DataFrame` should be a function that accepts similar arguments as pandas , and calls the actual constructor.

**Comment by floscha** [comment]
@thunterdb I guess you mean `ks.DataFrame` right? 😉 

@ueshin Sorry for being too vague. I was specifically referring to the case where a Spark DataFrame is used as the `data` argument. Then `index` will be used as a `Metadata` object and not accept a list as seen in the following code from *frame.py*:
```python
elif isinstance(data, spark.DataFrame):
    assert columns is None
    assert dtype is None
    assert not copy
    if index is None:
        super(DataFrame, self).__init__(_InternalFrame(data))
    else:
        super(DataFrame, self).__init__(_InternalFrame(
            data, data_columns=index.data_columns, index_map=index.index_map))
```
However, if you're already in the progress of refactoring `_InternalFrame`, then we can probably ignore this issue for now and see how things look once your refactoring is merged into the code base.

**Comment by ueshin** [comment]
@floscha Thanks for clarifying.
Yeah, I'm already in progress and planning to remove the whole `elif isinstance(data, spark.DataFrame)` clause within a few step-by-step PRs.

Btw, @thunterdb's idea that we should separate the `ks.DataFrame` function as a user-facing API and hide the actual constructor might be helpful. WDYT?
